src/arviz_stats/base/array.py

```diff
@@ -69,11 +73,17 @@ def hdi(
             "prob": prob,
             "skipna": skipna,
             "out_shape": (max_modes, 2) if method == "multimodal" else (2,),
+            "circular": circular,
         }
-        if method != "multimodal":
-            func_kwargs["circular"] = circular
-        else:
+        if method == "multimodal":
             func_kwargs["max_modes"] = max_modes
+            if is_discrete:
+                func_kwargs.pop("circular")
+                func_kwargs.pop("skipna")
+                func_kwargs["bins"] = None
+            else:
+                func_kwargs["bw"] = "isj" if not circular else "taylor"
+                func_kwargs.update(kwargs)
```

Comment:
Should this be de-indented one level? `_hdi_multimodal_discrete` has an `if bins is None` check but I don't see how `bins` can be different than `None` with discrete input.

Moreover, the `else` branch in this if ignores the value of `bins` and calls `_get_bins` which would be quite an unexpected behaviour. See comment below.

Reply:
I've updated to allow the user to provide `bins`, which was my original intention.